cpp/diffraction/StaticStructureFactorDebye.cc

```diff
@@ -86,7 +87,7 @@ void StaticStructureFactorDebye::accumulate(const freud::locality::NeighborQuery
                     // floating point precision errors can cause k to be
                     // slightly negative, and make evaluating the cylindrical
                     // bessel function impossible.
-                    auto abs_k = std::abs(k);
+                    auto abs_k = std::max(float(0.0), k);
```

Comment:
```suggestion
                    auto abs_k = std::fabs(k);
```
`fabs` is a more natural way of computing the absolute value.

Reply:
We don't really need the absolute value here. When I was originally writing the code, my first thought on how to prevent floating point error from creating very slightly negative numbers was to just take the absolute value. The way it's written now is correct, even though after looking at it again I've realized the variable name is a bit misleading.

Reply:
Right. Perhaps reassign `k` or name it `nonnegative_k`.